### DISCUSSION TITLE
Photon Machine Integration and Validation

### DISCUSSION THREAD

**Discussion by wahln** [e0404/matRad#772] [General]
This is a general discussion thread for Photon Machine Integration and Validation.

**Comment by wahln** [comment]
Some issues / threads concerned with this topic:
#769 , #767, #705

**Comment by edwardwang1** [comment]
Thanks Niklas for making this! I will summarize my experience here. We are using a Varian TrueBeam machine, and made a kernel for 6MV FFF.

**Generating Custom Kernel**
In order to validate the matRad dose calculation for a specific machine, it's necessary to have a machine specific pencil beam kernel. This can be done using the [repository ](https://github.com/e0404/photonPencilBeamKernelCalc) provided by the matRad authors. To create a kernel, the user needs to provide some input data. For us, we obtained this data from what was collected during machine commissioning at our centre. We also supplemented this data with virtual measurements in Eclipse (we wrote a script in ESAPI, which we can share if others want to do the same thing). This [issue](https://github.com/e0404/photonPencilBeamKernelCalc/issues/8) provides some information on the process from our perspective.  

**Validating Dose Calculation**
At our centre we use Eclipse for treatment planning. To validate our kernel, we are generating a plan in Eclipse, exporting the RTPlan and RTDose, and then importing them into matRad, and recalculating dose. This [issue](https://github.com/e0404/matRad/issues/769 ) summarizes our progress so far.

**Reply by wahln** [reply]
by the way - it would be interesting to see your validation work at our first user meeting - see #773

**Reply by edwardwang1** [reply]
Looks very cool! I will talk to my supervisor about this

**Reply by bouchraboukerroucha36-ctrl** [reply]
Hi @edwardwang1 @wahln  
I’m currently trying to add an Elekta machine to matRad, and I would really appreciate your advice.
Did you use MATLAB specifically for this workflow, or is it possible without it?
Also, if possible, could you share Eclipse parameters or commissioning settings that you used (beam model, kernel inputs, etc.)?
Seeing your Eclipse setup would help me adapt a similar process for Elekta.
Any tips or lessons learned would be very appreciated.

**Comment by edwardwang1** [comment]
Also, here are a number of phantoms that I generated with [dicomutils](https://github.com/raysearchlabs/dicomutils) that may be useful.

[CubePhantomDICOMs.zip](https://github.com/user-attachments/files/17063021/CubePhantomDICOMs.zip)

**Reply by wahln** [reply]
Nice, very useful.

**Comment by bouchraboukerroucha36-ctrl** [comment]
Hi, thank you for your help.
Could you explain how you vary or optimize the MLC ?
Also, how do you modify or edit structures in your workflow?
Any guidance or examples would be greatly appreciated.

**Reply by edwardwang1** [reply]
Hello,
Have you seen this [repo](https://github.com/e0404/photonPencilBeamKernelCalc/tree/master)? This is made by the matRad developers for generating custom pencil beam kernels, and contains sample data for a Synergy machine (which I think is Elekta?). 
This [paper](https://aapm.onlinelibrary.wiley.com/doi/10.1002/acm2.13811) has some instructions on how to do this as well.

The MLC optimization is done in the FMO and DAO. If you haven't already, I recommend going through a couple of the matRad tutorials.
